### PR TITLE
[tlgen] Unify Address Range for Device ports

### PR DESCRIPTION
**Pull request by eunchan**
With #1236 , Xbars now have `xbar_addr` field to indicate multiple addresses in a device port if the port is connected to another Xbar. `xbar_addr`, however, functions same as the `base_addr` and `size_byte` at the normal device port but having many of them.

It is logical to have unified data structure regardless of the device port's type (xbar vs non-xbar). This PR addresses the issue above. Now every device port has `addr_range` field, which type is a list of address group. The address group has `base_addr` and `size_byte` field in it.

### CODE REVIEW

File: util/tlgen/validate.py [hunk 1]
```diff
@@ -237,29 +248,30 @@ def validate(obj):  # OrderedDict -> Xbar
...
-            address_to = address_from + size - 1
+        #if node.node_type == NodeType.DEVICE and nodeobj["xbar"] == False:
+        #    # Add address obj["base_addr"], obj["size"])
+        #    node.xbar = False
+        #    address_from = int(nodeobj["base_addr"], 0)
+        #    size = int(nodeobj["size_byte"], 0)
+        #    address_to = address_from + size - 1
 
-            addr = (address_from, address_to)
+        #    addr = (address_from, address_to)
```
Comment (eunchan) [inline util/tlgen/validate.py]: Need to remove the code

File: util/tlgen/validate.py [hunk 2]
```diff
@@ -237,29 +248,30 @@ def validate(obj):  # OrderedDict -> Xbar
...
+        #    if checkAddressOverlap(addr, addr_ranges):
+        #        log.error(
+        #            "Address is overlapping. Check the config. Addr(0x%x - 0x%x)"
+        #            % (addr[0], addr[1]))
+        #        raise SystemExit("Address overlapping error occurred")
 
-            addr_ranges.append(addr)
-            node.addr_range = [addr]
+        #    addr_ranges.append(addr)
+        #    node.addr_range = [addr]
```
Comment (eunchan) [inline util/tlgen/validate.py]: Here also.. to be removed

File: util/tlgen/validate.py [hunk 3]
```diff
@@ -237,29 +248,30 @@ def validate(obj):  # OrderedDict -> Xbar
...
 
-        if node.node_type == NodeType.DEVICE and nodeobj["xbar"] == True:
-            node.xbar = True
+        #if node.node_type == NodeType.DEVICE and nodeobj["xbar"] == True:
```
Comment (eunchan) [inline util/tlgen/validate.py]: Another to be removed

File: util/topgen/merge.py [hunk 4]
```diff
@@ -421,6 +421,7 @@ def simplify_addr(addrs, xbar):
     @param addrs List of Dict[Addr] : {'base_addr':,'size_byte':}
     """
 
+    log.warning(addrs)
```
Comment (eunchan) [inline util/topgen/merge.py]: To be removed

File: hw/top_earlgrey/data/xbar_main.hjson [hunk 5]
```diff
@@ -80,17 +80,21 @@
       clock:     "clk_main_i",
       reset:     "rst_main_ni",
       inst_type: "rv_plic",
-      base_addr: "0x40090000",
-      size_byte: "0x1000",
+      addr_range: [{
```
Comment (msfschaffner) [inline hw/top_earlgrey/data/xbar_main.hjson]: nit: this jumps out a bit as it uses a different bracket placing as the other instances... maybe align the two styles...

File: hw/top_earlgrey/data/xbar_main.hjson [hunk 6]
```diff
@@ -80,17 +80,21 @@
...
       pipeline_byp: "false"
     },
     { name:      "pinmux",
       type:      "device",
       clock:     "clk_main_i",
       reset:     "rst_fixed_ni",
       inst_type: "pinmux",
-      base_addr: "0x40070000",
-      size_byte: "0x1000",
+      addr_range: [{
```
Comment (msfschaffner) [inline hw/top_earlgrey/data/xbar_main.hjson]: dito

**Pull request merged**